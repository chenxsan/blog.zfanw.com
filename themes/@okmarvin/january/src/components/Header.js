import React from 'react'
import PropTypes from 'prop-types'
import Menu from './Menu'
import { maxWidth, bps } from '../constants'
import { Link } from '@reach/router'
import { Col, Row, Inline } from 'jsxstyle'
const Header = props => (
  <Col
    flexShrink={0}
    alignItems='center'
    justifyContent='center'
    background='#fafafa'
    mediaQueries={bps}
    smFlexDirection='row'
    smBoxShadow='0 2px 0 0 #f2f2f2'
    smJustifyContent='space-between'
    smPaddingLeft={`calc((100% - ${maxWidth}) / 2)`}
    smPaddingRight={`calc((100% - ${maxWidth}) / 2)`}
    smBackground='#ffd54f'
  >
    <Row
      display='inline-flex'
      props={{ to: '/' }}
      component={Link}
      alignItems='center'
      color='#4a4a4a'
      fontSize='1.618em'
      lineHeight='2.625em'
      textDecoration='none'
      linkColor='#4a4a4a'
    >
      {props.logo && (
        <>
          <Inline
            component='img'
            height={42}
            fontSize={14}
            border='none'
            marginRight={10}
            props={{ src: props.logo, alt: 'logo' }}
          />
        </>
      )}
      {props.siteTitle}
    </Row>
    <Menu menu={props.menu} currentUrl={props.currentUrl} />
  </Col>
)
export default Header
Header.propTypes = {
  siteTitle: PropTypes.string.isRequired,
  logo: PropTypes.string,
  menu: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      permalink: PropTypes.string.isRequired
    })
  ),
  currentUrl: PropTypes.string.isRequired
}
Header.defaultProps = {
  menu: [],
  currentUrl: ''
}
