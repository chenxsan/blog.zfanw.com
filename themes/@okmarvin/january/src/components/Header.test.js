import React from 'react'
import Header from './Header'
import renderer from 'react-test-renderer'
it('renders correctly', () => {
  const tree = renderer.create(<Header siteTitle={'okmarvin'} />).toJSON()
  expect(tree).toMatchSnapshot()
})
