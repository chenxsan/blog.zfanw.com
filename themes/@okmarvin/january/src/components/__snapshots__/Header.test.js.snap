// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="_1hyv3hh"
>
  <a
    aria-current="page"
    className="_d7s2kr"
    href="/"
    onClick={[Function]}
  >
    okmarvin
  </a>
  <div
    className="_1u3u2z"
  />
</div>
`;
